### PR TITLE
docs: Added code review experiment proposal

### PR DESCRIPTION
**Pull request by BartoszJarocki**
This PR is a result of code review brainstorming sessions with @nickoferrall and @Dschoordsch.

The proposal comes from the tension that we're not iterating fast enough, especially when working on the new features. We spend a lot of time making our code perfect not even knowing if the feature we're working on will be useful to anyone. When things are not related to architecture or existing client-facing functionality we should be more comfortable with shipping not perfect code in exchange for being able to validate ideas/features faster. 

Let's discuss!

### CODE REVIEW

**Review by Dschoordsch — APPROVED** [review]

**Review by nickoferrall — APPROVED** [review]

**Review by enriquesanchez — APPROVED** [review]

**Review by ackernaut — APPROVED** [review]

**Review by JimmyLv — APPROVED** [review]

**Review by tianrunhe — APPROVED** [review]

File: docs/codeReview.md
Comment (jmtaber129) [inline docs/codeReview.md:47]: This feels like something that should need maintainer review _eventually_, since the feature flag will (hopefully) eventually be removed when the feature moves to GA.

No change needed for the experiment since maintainers will be reviewing these PRs later on anyway, but I think any future governance should address this in some way (e.g. "maintainer review not required to merge, but required before turning on the feature flag for more than X customers")

**Review by jmtaber129 — APPROVED** [review]

**Review by igorlesnenko — APPROVED** [review]

File: docs/codeReview.md
Comment (BartoszJarocki) [reply docs/codeReview.md:47]: It'd be hard to do the reviews after the code is already merged but what we can do is keep track of the PRs and manual tests that are needed to pass before removing the feature flag. The real intention is to move fast with things that don't require maintainer review in the first place ie. I'd still want to improve the way we split the work, submit smaller PRs, ask for feedback early and discuss all the architecture-related changes with the maintainer, preferably before doing the work. But yeah, let's see how it goes and learn as we go.

File: docs/codeReview.md
Comment (BartoszJarocki) [reply docs/codeReview.md:49]: thank you, fixed!

**Comment by BartoszJarocki** [comment]
> love it! this makes me really happy. reviews should just be a way to mitigate risk. if a PR has low risk, then the effort to review it should reflect that.
> 
> the only thing i'd add is about feature flag PRs: this might make for extra "review debt" for the PR that unlocks the feature. I'd like the author to keep track of some manual tests to run in that PR so we know specifically what pieces of the code haven't had as many eyes on it.
> 
> other than that, LGTM, let's do it! 🎉

I've added the following `It's the PR author's responsibility to keep track of the scenarios that need testing before removing the feature flag.`, let's see how it goes!

**Pull request merged**
